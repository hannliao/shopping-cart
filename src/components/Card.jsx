import '../styles/Card.css';

const Card = ({ imgSrc, title, price }) => {
  return (
    <div className="card">
      <img src={imgSrc} alt="" />
      <div className="info">
        <p>{title}</p>
        <p>${price}</p>
      </div>
    </div>
  );
};

export default Card;
