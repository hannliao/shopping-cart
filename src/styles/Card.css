.card {
  width: 300px;
  height: 400px;
  margin: 30px;
}

.card img {
  max-width: 100%;
  max-height: 100%;
}

.info {
  text-align: start;
  font-size: 14px;
}

.info p {
  color: gray;
}

.add {
  border: 1px solid black;
  width: 100%;
  padding: 15px;
  margin-top: 5px;
}
.add:hover {
  background-color: var(--lightgray);
}
