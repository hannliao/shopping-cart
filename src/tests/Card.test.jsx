import { vi, describe, it, expect } from 'vitest';
import { render, screen, waitFor } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Card from '../components/Card';

// vi.mock('../components/useProducts', () => ({
//   __esModule: true,
//   default: jest.fn(() => ({
//     products: [
//       {
//         id: 1,
//         imgSrc:
//           'https://cdn.shopify.com/s/files/1/0688/1755/1382/products/slides.jpg?v=1675447358',
//         title: 'Slides',
//         price: '$25',
//       },
//     ],
//     loading: false,
//     error: null,
//   })),
// }));

// describe('card', () => {
//   it('renders product image after loading', async () => {
//     render(
//       <BrowserRouter>
//         <Card />
//       </BrowserRouter>
//     );

//     await waitFor(() => {
//       expect(screen.getByText('Slides')).toBeInTheDocument();
//     });
//   });

// it('links to product detail page', () => {
//   render(
//     <BrowserRouter>
//       <Shop />
//     </BrowserRouter>
//   );
//   expect(screen.getByAltText('img', { name: Slides })).toHaveAttribute(
//     'href',
//     '/collections/shoes/products/slides'
//   );
// });
// });
