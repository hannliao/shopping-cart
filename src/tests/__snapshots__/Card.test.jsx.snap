// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`card > renders product image 1`] = `
<div>
  <div
    class="loading"
  >
    Loading...
  </div>
</div>
`;
